src/traces/contour/plot.js

```diff
@@ -103,6 +104,11 @@ function emptyPathinfo(contours, plotinfo, cd0) {
             z: cd0.z,
             smoothing: cd0.trace.line.smoothing
         });
+
+        if(pathinfo.length > 1000) {
+            Lib.warn('Too many contours, clipping at 1000', contours);
```

Comment:
somewhat arbitrary... but consistent with what we do for [cartesian ticks](https://github.com/plotly/plotly.js/blob/master/src/plots/cartesian/axes.js#L728) - here I included the warning as the consequences (in terms of super slow perf) are likely much more severe and you will want to see the `contours` object to know why it happened.